projects/distributed-tracing/src/pages/trace-detail/sequence/trace-sequence.component.ts

```diff
@@ -1,41 +1,40 @@
 import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { SubscriptionLifecycle } from '@hypertrace/common';
-
-import { Dashboard } from '@hypertrace/hyperdash';
 import { Observable } from 'rxjs';
-import { traceDetailDashboard } from '../trace-detail.dashboard';
-import { TraceDetails, TraceDetailService } from './../trace-detail.service';
+import { map } from 'rxjs/operators';
+import { TraceDetailService } from './../trace-detail.service';
+import { traceSequenceDashboard } from './trace-sequence.dashboard';
 
 @Component({
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
     <ht-navigable-dashboard
+      *htLoadAsync="this.traceVariables$ as traceVariables"
       class="scrollable-container"
       [padding]="0"
-      navLocation="${traceDetailDashboard.location}"
-      (dashboardReady)="this.onDashboardReady($event)"
+      [variables]="traceVariables"
+      navLocation="${traceSequenceDashboard.location}"
     >
     </ht-navigable-dashboard>
   `
 })
 export class TraceSequenceComponent {
-  public readonly traceDetails$: Observable<TraceDetails>;
-
-  public constructor(
-    private readonly subscriptionLifecycle: SubscriptionLifecycle,
```

Comment:
this was one problem - subscription lifecycle should be provided in the component whenever used, since it's meant to be be tied to the component lifecycle. With out an explicit provider here, we're not cleaning this up which is why it gets undefined in it when we navigate away.